src/enum-xyz.js

```diff
@@ -0,0 +1,32 @@
+/**
+ * @type {import("./types").enumOf}
+ * @param mapper should return the same value for the same name
+ */
+export const enumOf = (mapper, state) =>
+  new Proxy(
+    {},
+    {
+      // @ts-ignore
+      get: (_, name) => mapper(name, state),
+    }
+  )
+
+/**
+ * @type {import("./types").memoEnumOf}
+ * @param mapper always returns the same value for the same name
+ */
+export const memoEnumOf = (mapper) =>
```

Comment:
Same with this!

Reply:
However it's guaranteed here. A great example is `Symbol("X"`) != Symbol("X")`. This makes sure `Symbols.X === Symbols.X`!